src/sysrepo.c

```diff
@@ -4266,10 +4266,21 @@ sr_apply_oper_changes(struct sr_mod_info_s *mod_info, sr_session_ctx_t *session,
     }
 
 cleanup:
+    /* remove any previous oper_push_cache */
+    lyd_free_siblings(session->oper_push_cache);
+    session->oper_push_cache = NULL;
+
+    /* store mod_info->data as the new oper_push_cache only if there are no errors */
+    if (!err_info && !*err_info2 && !(session->conn->opts & SR_CONN_NO_OPER_PUSH_CACHE)) {
+        session->oper_push_cache = mod_info->data;
+        mod_info->data = NULL;
+    }
+
```

Comment:
Why is this code in the cleanup? I think freeing the cache could occur only on success (so can be moved before the label, on error the old cache can remain) and similarly updating the cache.